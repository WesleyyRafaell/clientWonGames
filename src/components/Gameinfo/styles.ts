import styled, { css } from 'styled-components'

import * as RibbonStyles from 'components/Ribbon/styles'
import * as Button from 'components/Button/styles'

import media from 'styled-media-query'

export const Wrapper = styled.main`
  ${({ theme }) => css`
    width: 100%;
    max-width: 125.3rem;
    background-color: ${theme.colors.white};
    padding: ${theme.spacings.medium} ${theme.spacings.small};
    position: relative;

    ${Button.Wrapper} {
      margin-bottom: ${theme.spacings.xsmall};
    }

    ${media.greaterThan('medium')`
      padding: ${theme.spacings.medium} ${theme.spacings.large};

      ${RibbonStyles.Wrapper} {
        right: ${theme.spacings.large};
        top: ${theme.spacings.small};
        font-size: ${theme.font.sizes.large};
        &:before {
          border: none;
        }
      }
    `}
  `}
`

export const ButtonsContainer = styled.div`
  ${media.greaterThan('medium')`
    display: flex;
    flex-direction: row-reverse;
    justify-content: end;

    ${Button.Wrapper} {
      width: 18.4rem
    }
  `}
`

export const Text = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.gray};
    line-height: 2.1rem;
    margin-bottom: ${theme.spacings.small};
    max-width: 66.3rem;
  `}
`
