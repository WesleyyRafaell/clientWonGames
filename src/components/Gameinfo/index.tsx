import Button from 'components/Button'
import Heading from 'components/Heading'
import Ribbon from 'components/Ribbon'
import {
  AddShoppingCart,
  FavoriteBorder
} from '@styled-icons/material-outlined'

import * as S from './styles'

export type GameInfoProps = {
  title: string
  price: string
  description: string
}

const Gameinfo = ({ title, price, description }: GameInfoProps) => (
  <S.Wrapper>
    <Ribbon color="secondary">R$ {price}</Ribbon>
    <Heading color="black" lineBottom lineColor="primary">
      {title}
    </Heading>
    <S.Text>{description}</S.Text>
    <S.ButtonsContainer>
      <Button icon={<AddShoppingCart />} fullWidth>
        Add to cart
      </Button>
      <Button icon={<FavoriteBorder />} fullWidth minimal>
        Wishlist
      </Button>
    </S.ButtonsContainer>
  </S.Wrapper>
)

export default Gameinfo
