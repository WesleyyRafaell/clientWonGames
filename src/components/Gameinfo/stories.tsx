import { Story, Meta } from '@storybook/react'
import Gameinfo, { GameInfoProps } from '.'

const mockGame = {
  title: 'Borderlands 3',
  price: '215,00',
  description:
    'Experience the epic space strategy games that redefined the RTS genre. Control your fleet and build an armada across more than 30 single-player missions.'
}

export default {
  title: 'Game/Gameinfo',
  component: Gameinfo,
  parameters: {
    backgrounds: {
      default: 'won-dark'
    }
  },
  args: mockGame
} as Meta

export const Default: Story<GameInfoProps> = (args) => <Gameinfo {...args} />
