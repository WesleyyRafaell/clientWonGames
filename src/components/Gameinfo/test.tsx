import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helper'

import Gameinfo from '.'

const props = {
  title: 'My Game Title',
  description: 'Game Description',
  price: '210,00'
}

describe('<Gameinfo />', () => {
  it('should render game informations', () => {
    renderWithTheme(<Gameinfo {...props} />)

    expect(
      screen.getByRole('heading', { name: /My Game Title/i })
    ).toBeInTheDocument()

    expect(screen.getByText('Game Description')).toBeInTheDocument()
    expect(screen.getByText('R$ 210,00')).toBeInTheDocument()
  })

  it('should render buttons', () => {
    const { container } = renderWithTheme(<Gameinfo {...props} />)

    expect(
      screen.getByRole('button', { name: /Add to cart/i })
    ).toBeInTheDocument()

    expect(
      screen.getByRole('button', { name: /wishlist/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
