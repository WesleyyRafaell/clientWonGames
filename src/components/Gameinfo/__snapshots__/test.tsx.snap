// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Gameinfo /> should render buttons 1`] = ` `
